### PR TITLE
[Issue #370]Word2Vec (WorkInProgress)

### PR DESCRIPTION
**Pull request by psyyz10**
For issue #370

### CODE REVIEW

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/LowerCase.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (fanshiqing) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/LowerCase.scala]: License: ASF => Intel Coporation
The same as below.

Comment (psyyz10) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/LowerCase.scala]: Changed, and thanks a lot :+1:

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/Tokenizer.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @param sep the delimiting regular expression to split on.
+ *            Defaults to matching all punctuation and whitespace
+ */
+class Tokenizer(sep: String) extends Transformer[String, Seq[String]] {
+  override def apply(prev: Iterator[String]): Iterator[Seq[String]] =
+    prev.map(x => x.split(sep))
+}
+
+object Tokenizer {
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/Tokenizer.scala]: BTW, spark mllib also have a tokenizer...

https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala

Comment (psyyz10) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/Tokenizer.scala]: Yan's version is not exactly the same as I want. Zhichao's version is similar to what I want, it is possible to combine to one, I will discuss with him. Thx.

Comment (psyyz10) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/Tokenizer.scala]: Thanks for reminder, spark mllib's RegexTokenizer is exactly what I want, maybe I need to see how to use it and insert to the transformer pipeline

**Comment by linbojin** [comment]
Do we still want to add Word2Vec?

**Comment by psyyz10** [comment]
@linbojin This PR's priority is low, and the network is shallow so not suitable for Synchronize SGD with all-reduce. Spark MLlib's word2vec can be used as well with the same function. Do you have any requirement on this PR?

**Pull request closed without merging**